Advanced-Recruit-Training/src/BatteryStateOfChargeService.cpp

```diff
@@ -16,21 +21,58 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+    return totalAmpHoursUsed_;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
-    return false;
+    if (current_ < 0)
+    {
+       return true;
+    }
+    else
+    {
+        return false;
+    }
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 {
-    return QTime::currentTime();
-}
+    int ms_remaining;
+    if (isCharging())
+    {
+      ms_remaining = totalAmpHoursUsed() / averageCurrent_ * HOURS_TO_MS_CONVERSION;
+    }
+    else
+    {
+      ms_remaining = (BATTERY_AMP_HOUR_CAPACITY - totalAmpHoursUsed()) / averageCurrent_ * HOURS_TO_MS_CONVERSION;
+    }
+
+    return QTime(0, 0, 0, 0).addMSecs(ms_remaining);
 
+}
+/*running average needs to be just the one after it, so I need to write the average as the current + past current divided by 2*/
```

Comment:
For comments meant to be seen by others, avoid the use of "I". It sounds like a mental note instead of one meant to be read.

I would rewrite this to `The running average of the current should be the (current + pastcurrent)/2`. 
Also, comments here should be more descriptive of the overall function, rather than describing a part of it. Consider putting it right above line 67 instead.